hdf5-sys/build.rs

```diff
@@ -584,27 +593,123 @@ impl Config {
         }
         if self.header.have_direct {
             println!("cargo:rustc-cfg=h5_have_direct");
+            println!("cargo:have_direct=1");
         }
         if self.header.have_parallel {
             println!("cargo:rustc-cfg=h5_have_parallel");
+            println!("cargo:have_parallel=1");
         }
         if self.header.have_threadsafe {
             println!("cargo:rustc-cfg=h5_have_threadsafe");
+            println!("cargo:have_threadsafe=1");
         }
     }
 }
 
 fn main() {
-    let mut searcher = LibrarySearcher::new_from_env();
-    searcher.try_locate_hdf5_library();
-    let config = searcher.finalize();
-    println!("{:#?}", config);
-    config.emit_link_flags();
+    if feature!("STATIC").is_ok() {
+        build_and_emit();
+    } else {
+        let mut searcher = LibrarySearcher::new_from_env();
+        searcher.try_locate_hdf5_library();
+        let config = searcher.finalize();
+        println!("{:#?}", config);
+        config.emit_link_flags();
+        config.emit_cfg_flags();
+    }
+}
+
+fn build_and_emit() {
+    println!("cargo:rerun-if-changed=build.rs");
+    let mut hdf5config = cmake::Config::new("source");
+    hdf5config
+        .define("HDF5_NO_PACKAGES", "ON")
+        .define("BUILD_SHARED_LIBS", "OFF") // Only need to build a static library
+        .define("BUILD_TESTING", "OFF")
+        .define("HDF5_BUILD_TOOLS", "OFF")
+        .define("HDF5_BUILD_EXAMPLES", "OFF")
+        //
+        .define("HDF5_BUILD_JAVA", "OFF")
+        .define("HDF5_BUILD_FORTRAN", "OFF")
+        .define("HDF5_BUILD_CPP_LIB", "OFF")
+        // Default, with feature flags
+        .define("HDF5_ENABLE_PARALLEL", "OFF")
+        .define("HDF5_ENABLE_DEPRECATED_SYMBOLS", "OFF")
+        .define("HDF5_ENABLE_THREADSAFE", "OFF")
+        .define("ALLOW_UNSUPPORTED", "OFF")
+        .define("HDF5_BUILD_HL_LIB", "OFF");
+
+    if feature!("ZLIB").is_ok() {
+        let zlib_include_dir = std::env::var_os("DEP_Z_INCLUDE").unwrap();
+        let mut zlib_header = std::env::split_paths(&zlib_include_dir).nth(0).unwrap();
+        zlib_header.push("zlib.h");
+        let zlib_lib = "z";
+        // libz-sys does not emit a link flag, so we emit one here
+        println!("cargo:rustc-link-lib=static={}", &zlib_lib);
+        hdf5config
+            .define("HDF5_ENABLE_Z_LIB_SUPPORT", "ON")
+            .define("H5_ZLIB_HEADER", zlib_header)
+            .define("ZLIB_STATIC_LIBRARY", &zlib_lib);
+    }
+
+    if feature!("DEPRECATED").is_ok() {
+        hdf5config.define("HDF5_ENABLE_DEPRECATED_SYMBOLS", "ON");
+    }
+
+
+    if feature!("THREADSAFE").is_ok() {
+        hdf5config.define("HDF5_ENABLE_THREADSAFE", "ON");
+        if feature!("HL").is_ok() || feature!("PARALLEL").is_ok() {
+            println!("cargo:warning=Unsupported selection of items. This is probably ok");
+            hdf5config.define("ALLOW_UNSUPPORTED", "ON");
+        }
+    }
+
+    if feature!("HL").is_ok() {
```

Comment:
Does hl make sense given that we don't provide bindings for it at this point?

Reply:
Both `hl` and `deprecated` are included to reuse the library in the `netcdf` crate. These features are only needed to get the right build flags and linking. Since these are hidden behind feature flags on `hdf5-sys`, I would say they are obscured enough that a user would not expect them to be available.